modules/text2colbert-jinaai/ent/class_settings.go

```diff
@@ -0,0 +1,115 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package ent
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/entities/schema"
+	basesettings "github.com/weaviate/weaviate/usecases/modulecomponents/settings"
+)
+
+const (
+	// Default values for URL (model is ok) cannot be changed before we solve how old classes that have the defaults
+	// NOT set will handle the change
+	DefaultJinaAIModel           = "jina-colbert-v2"
+	DefaultVectorizeClassName    = true
+	DefaultPropertyIndexed       = true
+	DefaultVectorizePropertyName = false
+	DefaultBaseURL               = "https://api.jina.ai"
+	LowerCaseInput               = false
+)
+
+var DefaultDimensions int64 = 128
+
+type classSettings struct {
+	basesettings.BaseClassSettings
+	cfg moduletools.ClassConfig
+}
+
+func NewClassSettings(cfg moduletools.ClassConfig) *classSettings {
+	return &classSettings{cfg: cfg, BaseClassSettings: *basesettings.NewBaseClassSettings(cfg, LowerCaseInput)}
+}
+
+func (cs *classSettings) Model() string {
+	return cs.getProperty("model", DefaultJinaAIModel)
+}
+
+func (cs *classSettings) BaseURL() string {
+	return cs.getProperty("baseURL", DefaultBaseURL)
+}
+
+func (cs *classSettings) Dimensions() *int64 {
+	return cs.BaseClassSettings.GetPropertyAsInt64("dimensions", &DefaultDimensions)
+}
+
+func (cs *classSettings) Validate(class *models.Class) error {
+	if cs.cfg == nil {
+		// we would receive a nil-config on cross-class requests, such as Explore{}
+		return errors.New("empty config")
+	}
+
+	if err := cs.BaseClassSettings.ValidateClassSettings(); err != nil {
+		return err
+	}
+
+	err := cs.validateIndexState(class)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cs *classSettings) getProperty(name, defaultValue string) string {
+	return cs.BaseClassSettings.GetPropertyAsString(name, defaultValue)
+}
+
+func (cs *classSettings) validateIndexState(class *models.Class) error {
```

Comment:
the baseclassSettings.Validate() function already does this

Reply:
yeah! forgot about this, good catch!